src/decider/agent/copyrightDeactivationClutterRemovalScript.py

```diff
@@ -0,0 +1,187 @@
+                    if THE_PROBABLE_LOGIC_NER_CHECK[0] in _values:
+                        df.loc[index,'is_copyright'] = "t"
+                    elif THE_PROBABLE_LOGIC_NER_CHECK[2] in _values:
+                        for _val in listB["Entity"]:
+                                pattern_regex = r'''((?:19|20)\d{2}|\d{2})(?!\d)(?:[, \t-]{1,3}((?:19|20)\d{2}|\d{1,2}))?'''
+                                extract_list = re.search(pattern_regex,_val)
+                                if extract_list:
+                                    df.loc[index,'is_copyright'] = "t"
+                    elif THE_PROBABLE_LOGIC_NER_CHECK[1] in _values or THE_PROBABLE_LOGIC_NER_CHECK[3] in _values:
+                        if THE_PROBABLE_LOGIC_NER_CHECK[0] in _values:
+                            df.loc[index,'is_copyright'] = "t"
+        else:
+            df.loc[index,'is_copyright'] = "f"
+        
+        if clutter_flag == 1:
+            if df.loc[index,'is_copyright'] == "t":
+                clutterRemoval(df, index, listB)
+    return
+
+def clutterRemoval(df, index, ner_list):
+    string1 = "all rights reserved"
+    string2 = "distributed under the mit software license"
+    string3 = df.loc[index,'content']
+
+    if string1 in string3:
+        clutter_removed = string3[:string3.index(string1)]
+        df.loc[index,'edited_text'] = clutter_removed 
+
+    elif string2 in string3:
+        clutter_removed = string3[:string3.index(string1)]
+        df.loc[index,'edited_text'] = clutter_removed
+    
+    elif 'ORG' in ner_list['Values'] and 'PERSON' in ner_list['Values']:
+        org_name = ner_list['Entity'][ner_list['Values'].index('ORG')]
+        person_name = ner_list['Entity'][ner_list['Values'].index('PERSON')]
+        try:
+            if string3.index(org_name) > string3.index(person_name):
+                clutter_removed = string3[:string3.index(org_name)] + org_name
+                df.loc[index,'edited_text'] = clutter_removed
+            else:    
+                clutter_removed = string3[:string3.index(person_name)] + person_name
+                df.loc[index,'edited_text'] = clutter_removed
+        except:
+            df.loc[index,'edited_text'] = string3
+
+    elif 'ORG' in ner_list['Values']:
+        org_name = ner_list['Entity'][ner_list['Values'].index('ORG')]
+        try:
+            string3 = re.sub(r'''[^\w\s()]+''', '', string3)
+            clutter_removed = string3[:string3.index(org_name)] + org_name
+            df.loc[index,'edited_text'] = clutter_removed
+        except:
+            df.loc[index,'edited_text'] = string3
+
+    elif 'PERSON' in ner_list['Values']:
+        person_name = ner_list['Entity'][ner_list['Values'].index('PERSON')]
+        try:
+            clutter_removed = string3[:string3.index(person_name)] + person_name
+            df.loc[index,'edited_text'] = clutter_removed
+        except:
+            df.loc[index,'edited_text'] = string3
```

Comment:
This as a whole can be simplified to:

```python
org_pos = -1
person_pos = -1
if 'ORG' in ner_list['Values']:
  string3 = re.sub(r'''[^\w\s()]+''', '', string3) # I am not sure why this is in only one of the branch
  las_ent = len(ner_list['Values']) - ner_list['Values'][::-1].index('ORG') - 1 # Get the last org
  org_name = ner_list['Entity'][las_ent]
  try:
    org_pos = string3.rindex(org_name) + len(org_name)
  except:
    org_pos = -1
if 'PERSON' in ner_list['Values']:
  las_ent = len(ner_list['Values']) - ner_list['Values'][::-1].index('PERSON') - 1 # Get the last person
  person_name = ner_list['Entity'][las_ent]
  try:
    person_pos = string3.rindex(person_name) + len(person_name)
  except:
    person_pos = -1

if org_pos > 0 or person_pos > 0:
  del_upto = max(org_pos, person_pos)
  df.loc[index,'edited_text'] = string3[:del_upto]
```

Note, I've modified the script to get last person from the list and last org, also to get the last occurrence of them in the original copyright to prevent as much data loss as possible. If the results are not as intended, please revert them.